### DISCUSSION TITLE
What is uername and password?

### DISCUSSION THREAD

**Discussion by zinwelzl** [pablosnt/rekono#37] [Help] [answered]
Docker installation, works great.

https://github.com/pablosnt/rekono/blob/main/docker/README.md#initial-rekono-user

user:rekono
password:rekono

This don't work!

Linux Mint latest.

**Comment by pablosnt** [comment]
This credentials should work in Docker environment. However, there were more issues like that, so I will review and try to reproduce this problem. Can you tell me if the directory where you clone Rekono is a non-root directory? The issue #33 has been resolved retrying from a non-root directory after remove all Docker images and volumes.

**Comment by zinwelzl** [comment]
Linux Mint latest, updated, in VirtualBox.
Non-root directory and non-root user.
What should I do?
Delete all images and volumes, and install again?

**Comment by zinwelzl** [comment]
images and volumes removed

docker-compose up -d
[+] Running 0/12
 ⠿ initialize Error                                                                                                                                   2.1s
 ⠿ frontend Error                                                                                                                                     2.1s
 ⠿ telegram-bot Error                                                                                                                                 2.1s
 ⠿ findings-worker Error                                                                                                                              2.1s
 ⠿ postfix Error                                                                                                                                      2.1s
 ⠿ backend Error                                                                                                                                      2.1s
 ⠿ postgres Error                                                                                                                                     2.1s
 ⠿ redis Error                                                                                                                                        2.1s
 ⠿ emails-worker Error                                                                                                                                2.1s
 ⠿ tasks-worker Error                                                                                                                                 2.1s
 ⠿ executions-worker Error                                                                                                                            2.1s
 ⠿ nginx Error                                                                                                                                        2.1s
Error response from daemon: pull access denied for rekono-backend, repository does not exist or may require 'docker login': denied: requested access to the resource is denied

**Comment by pablosnt** [comment]
I think that you didn't run `docker-compose build` first. I will review the problem carefully  in your platform. By the time, please retry. As I said, it should works

**Comment by zinwelzl** [comment]
I did run build first, but still can't login.

**Comment by pablosnt** [comment]
Sorry, I have tried to reproduce the error in Linux Mint and I can't do it. Please, tell me if I'm doing something wrong:

OS information in VirtualBox:

<img width="1175" alt="Captura de Pantalla 2022-09-10 a las 16 53 05" src="https://user-images.githubusercontent.com/69458381/189488975-019c2de4-aaa2-449d-bd9e-8dc04112204f.png">


After execute `docker-compose build` and `docker-compose up -d`:

<img width="546" alt="Captura de Pantalla 2022-09-10 a las 16 50 41" src="https://user-images.githubusercontent.com/69458381/189488980-867480d1-01a6-4c28-8352-5937c395e05e.png">


Finally, I log in with default credentials `rekono:rekono` in the UI:

<img width="1195" alt="Captura de Pantalla 2022-09-10 a las 16 53 29" src="https://user-images.githubusercontent.com/69458381/189488987-a21cda48-0aef-4e07-bd20-e82e238234b8.png">

**Comment by pablosnt** [comment]
I advise to check the following items to prevent or get more information about the error:

- Check if the default user has been created in the initialize log.
- Check the API response using the browser development options: if the request arrives the backend and if credentials are wrong, the response should be 401.
- Check if the authentication requests arrives the Rekono API in the backend log.
- If requests don't arrive the backend, please reboot the nginx container because it doesn't forward the requests correctly.

Please, check this to know more about the error, and maybe fix it. And if you need, provide me this information to been able to help you.

**Comment by zinwelzl** [comment]
Nginx container was problem, after restart login work now!

Thanks for your help, great tool you build!

**Comment by zinwelzl** [comment]
BTW, just to let you know.

I did try few more Mint VM and every time same problem, and same solution with Nginx restart.

But with Kali, Rekono works great with first install and also in my cloud server.

**Reply by pablosnt** [reply]
Hello, after a more deep investigation, I was able to reproduce the error in Ubuntu and I think that the reason of the error could be the docker-compose version. In my local machine I have the following one and the dependencies between the containers are respected:
<img width="451" alt="Captura de Pantalla 2022-09-20 a las 10 43 01" src="https://user-images.githubusercontent.com/69458381/191212981-5ee58ded-0189-4652-9f1b-6d9b88f8d04b.png">
Otherwise, the error appears when I execute Rekono in Ubuntu with docker-compose 1.29.2 (https://github.com/pablosnt/rekono/actions/runs/3088518948/jobs/4995087093#step:3:4210). In the `docker-compose logs` you can check that the backend container is created before the database. So please, check your docker-compose version and upgrade it if needed. Let me know your conclusions, thank you!

**Reply by zinwelzl** [reply]
docker-compose version 1.29.2, build unknown

Same here, after upgrade rekono works!

**Comment by pablosnt** [comment]
Perfect!!! Thank you for keep trying. Your feedback about Linux Mint will be very useful to investigate future issues.

**Comment by pablosnt** [comment, marked as answer]
At the moment, the conclusions of this problem are:

- **Upgrade your `docker-compose --version`. It has been tested that the Docker environment fails with versions like 1.29.2**
- Check if the requests reach the **backend** container using the `docker-compose logs`. If not, reboot your **nginx** container to prevent forwarding errors
- Ensure that the default user has been created correctly in the **initialize** container using the `docker-compose logs`
